sql/hive/src/main/scala/org/apache/spark/sql/hive/MetastoreRelation.scala

```diff
@@ -95,8 +95,15 @@ private[hive] case class MetastoreRelation(
     val (partCols, schema) = catalogTable.schema.map(toHiveColumn).partition { c =>
       catalogTable.partitionColumnNames.contains(c.getName)
     }
+
+    // keep the schema order with catalogTable.partitionColumnNames
+    val reorderPartCols = catalogTable.partitionColumnNames.flatMap {
```

Comment:
create table has a reorderd partcols
https://github.com/windpiger/spark/blob/4612a52e7e1adb7ce41fe8329ad67faa11285550/sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/CreateHiveTableAsSelectCommand.scala#L87

so here is redundant, we can remove it.